examples/federated-learning/func_lib.py

```diff
@@ -0,0 +1,63 @@
+import tensorflow as tf
+import tf_encrypted as tfe
+
+
+### Example model_fns ###
+
+def default_model_fn(data_owner):
+  """Runs a single training step!"""
+  x, y = next(data_owner.dataset)
+
+  with tf.name_scope("gradient_computation"):
+
+    with tf.GradientTape() as tape:
+      preds = data_owner.model(x)
+      loss = tf.reduce_mean(data.loss(y, preds, from_logits=True))
+
+    grads = tape.gradient(loss, data_owner.model.trainable_variables)
+
+  return grads
+
+
+def reptile_model_fn(data_owner, iterations=3, grad_fn=default_model_fn, **kwargs):
+  # This corresponds to the Reptile variant that computes k steps of SGD.
+  # When paired with the secure_aggregation aggregator_fn, this model_fn
+  # corresponds to using g_k as the outer gradient update. See the Reptile
+  # paper for more: https://arxiv.org/abs/1803.02999
+  for k in range(iterations):
+    grads_k = grad_fn(data_owner, **kwargs)
+    data_owner.optimizer.apply_gradients(
+        zip(grads_k, data_owner.model.trainable_variables),
+    )
+
+  return model.trainable_variables
```

Comment:
just making sure that it's okay to use weights as gradients here..?

Reply:
See the last paragraph in Section 3 of the [Reptile paper](https://arxiv.org/abs/1803.02999), which points to Algorithm 3 in [this NeurIPS paper](https://papers.nips.cc/paper/4006-parallelized-stochastic-gradient-descent) as the preferred way to parallelize Reptile.

Now that you raise this point though, there should be a step where we subtract the averaged weights from the current master model's weights, so this is definitely a bug in the current implementation.